transforms/images/polus-imagej-filter-gauss-plugin/plugin.json

```diff
@@ -0,0 +1,81 @@
+{
+  "name": "ImageJ filter gauss",
+  "version": "0.3.0",
+  "title": "ImageJ filter gauss",
+  "description": "This plugin applies a Gaussian Convolutional filter to an input collection.",
+  "author": "Benjamin Houghton (benjamin.houghton@axleinfo.com)",
+  "institution": "National Center for Advancing Translational Sciences, National Institutes of Health",
+  "repository": "https://github.com/labshare/polus-plugins",
+  "website": "https://ncats.nih.gov/preclinical/core/informatics",
+  "citation": "",
+  "containerId": "polusai/imagej-filter-gauss-plugin:0.3.0",
+  "inputs": [
+    {
+      "name": "opName",
+      "type": "enum",
+      "options": {
+        "values": ["GaussRAISingleSigma",
+                    "DefaultGaussRAI"]
+      },
+      "description": "Operation overloading method to perform",
+      "required": false
+    },
+    {
+      "name": "inpDir",
+      "type": "collection",
+      "description": "Collection to be processed by this plugin",
+      "required": false
+    },
+    {
+      "name": "sigma",
+      "type": "number",
+      "description": "The standard deviation of the Gaussian filter, same in both dimensions.",
+      "required": false
+    },
+    {
+      "name": "sigmas",
+      "type": "array",
+      "description": "The standard deviations of the Gaussian filters, (x,y)",
+      "required": false,
+      "options":{
+        "items": {
+          "type": "string",
+          "title": "The standard deviations of the Gaussian filters, (x,y)",
+          "widget": "string"
+        }
+      }
+    }
+  ],
+  "outputs": [
+    {
+      "name": "outDir",
+      "type": "collection",
+      "description": "Output collection"
+    }
+  ],
+  "ui": [
+    {
+      "key": "inputs.opName",
+      "title": "opName",
+      "description": "Operation overloading method to perform"
+    },
+    {
+      "key": "inputs.inpDir",
+      "title": "inpDir",
+      "description": "Collection to be processed by this plugin",
+      "condition": "model.inputs.opName==GaussRAISingleSigma or model.inputs.opName==DefaultGaussRAI"
```

Comment:
This condition will always evaluate to True. Why even have it? Is it difficult to generalize in the pipeline?

Reply:
> It looks like these two arguments are always required. Should the required fields be set to True? There might be some WIPP shenanigans that I'm not aware of that make the current settings necessary.

These are practically the same issue/problem. To solve this additional code would need to be added to the pipeline to check for inputs that are required for all overloading method. While I don't think it would be overly difficult it would take some modification to file that parses the op information, `populate.py` and it has not been a very high priority as there is a lot of improvements which need to be made to the entire pipeline. This is something I should work to correct in future iterations of the pipeline. For this plugin I have removed this condition.